packages/library-base/src/commonMain/kotlin/io/realm/schema/RealmClass.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.schema
+
+interface RealmClass { // Matches realm_class_info_t except that attributes are also attached here
+    val name: String
+    val properties: Collection<RealmProperty>
```

Comment:
Any thinking behind using Collection here? would maybe expect List or Set to be more appropriate?

Reply:
Sets doesn't offer random access apis while lists doesn't signal uniqueness of elements. But reiterating it I think that I removed the random access apis, so I guess we could just go with sets again. 

I don't know if there are different clear expectations between sets/list that would be nice to keep in mind (constant order, etc.) and if it is only me being annoyed about using List if we guarantee uniqueness of elements?